python_modules/libraries/dagster-components/dagster_components/lib/example_custom_component.py

```diff
@@ -0,0 +1,29 @@
+from dagster import Definitions
+from pydantic import BaseModel, TypeAdapter
+from typing_extensions import Self
+
+from dagster_components import ComponentLoadContext, component
+from dagster_components.core.component_decl_builder import ComponentDeclNode, YamlComponentDecl
+from dagster_components.lib.custom_component import CustomComponent
+
+
+class ExampleCustomComponentParams(BaseModel): ...
+
+
+@component(name="example_custom_component")
+class ExampleCustomComponent(CustomComponent):
```

Comment:
Confused why we are exposing this as a registered component. If the user generates a custom component they get basically the same code.

Reply:
It's so we can invoke `dg generate`

A cleaner separation between component generation and the runtime component would make this clearer. I think this is a good motivation use case for that.

Reply:
actually nvm you are correct this makes no sense